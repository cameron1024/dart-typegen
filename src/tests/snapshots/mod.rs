// #[test]
// fn test_foo() {
//     let dart = include_str!("./foo.dart");
//     let toml = include_str!("./foo.toml");
//
// }
