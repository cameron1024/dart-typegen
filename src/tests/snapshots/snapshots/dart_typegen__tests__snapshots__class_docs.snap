---
source: src/tests/snapshots/mod.rs
expression: dart
---
import "package:equatable/equatable.dart";

/// Single line docs
final class Foo with EquatableMixin {
  final String x;

  const Foo({required this.x});

  @override
  List<Object?> get props => [x];

  FooBuilder toBuilder() => FooBuilder(x: x);

  Map<String, dynamic> toJson() => {"x": x};
  factory Foo.fromJson(Map<String, dynamic> json) =>
      Foo(x: json["x"] as String);

  @override
  String toString() =>
      "Foo("
      "x: $name"
      ")";
}

/// Builder class for [Foo]
final class FooBuilder {
  String x;

  FooBuilder({required this.x});

  Foo build() => Foo(x: x);
}

/// Multi line docs
///
/// More stuff
final class Bar with EquatableMixin {
  final String x;

  const Bar({required this.x});

  @override
  List<Object?> get props => [x];

  BarBuilder toBuilder() => BarBuilder(x: x);

  Map<String, dynamic> toJson() => {"x": x};
  factory Bar.fromJson(Map<String, dynamic> json) =>
      Bar(x: json["x"] as String);

  @override
  String toString() =>
      "Bar("
      "x: $name"
      ")";
}

/// Builder class for [Bar]
final class BarBuilder {
  String x;

  BarBuilder({required this.x});

  Bar build() => Bar(x: x);
}
